chips/apollo3/src/chip.rs

```diff
@@ -3,31 +3,27 @@
 use core::fmt::Write;
 use cortexm4;
 use kernel::Chip;
+use kernel::Platform;
 
-use crate::ble;
-use crate::gpio;
-use crate::iom;
-use crate::nvic;
-use crate::stimer;
-use crate::uart;
-
-pub struct Apollo3 {
+pub struct Apollo3<P: Platform + 'static> {
     mpu: cortexm4::mpu::MPU,
     userspace_kernel_boundary: cortexm4::syscall::SysCall,
     scheduler_timer: cortexm4::systick::SysTick,
+    platform: &'static P,
 }
 
-impl Apollo3 {
-    pub unsafe fn new() -> Apollo3 {
-        Apollo3 {
+impl<P: Platform + 'static> Apollo3<P> {
```

Comment:
For `earlgrey`, I think it is worth now initialise the `chip_config::Config, from chip_config.rs` in the corresponding `board` and passing the structure as a parameter into `new`.

For OpenTitan it will also allow to remove the following features from the `.../earlgrey/Cargo.toml`:
```
[features]
config_fpga_nexysvideo = ["config_disable_default"]
config_sim_verilator = ["config_disable_default"]
config_disable_default = []
```
And instead have the config related features entirely in the board Cargo.toml.

P.S.

I know this is `apollo` related change, but seems like the most appropriate place to leave this comment.